src/lib.rs

```diff
@@ -1,564 +1,570 @@
-        let mut reader = ValueReader::new(&path)?;
+// This wrapper is needed to avoid lifetime problems
+#[derive(Debug, Clone)]
+pub(crate) struct SequenceRecords(Vec<SequenceRecord>);
+
+impl SequenceRecords {
+    #[inline]
+    pub fn query(&self, chr: impl AsRef<str>) -> Result<&SequenceRecord> {
+        let chr = chr.as_ref();
+        self.0
+            .iter()
+            .find(|seq| seq.chr == chr)
+            .ok_or_else(|| Error::MissingName(chr.to_string()))
+    }
+}
 
-        let mut sequences = HashMap::new();
+impl Deref for SequenceRecords {
+    type Target = [SequenceRecord];
 
-        let sequence_count = reader.field()?;
-        let _reserved = reader.field(); // read the unused reserved field
+    #[inline]
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
 
-        for _ in 0..sequence_count {
-            let name_size = reader.byte()? as usize;
-            let name = reader.string(name_size)?;
-            let seq_offset = reader.field()?;
+/// Information on a particular chromosome or sequence in a 2bit file.
+#[derive(Debug, Clone, PartialEq)]
+pub struct SequenceInfo {
+    /// Chromosome or sequence name
+    pub chr: String,
+    /// Number of nucleotides in the sequence
+    pub length: usize,
+    /// Total length of all hard masks
+    pub hard_masks_total_length: usize,
+    /// Total length of all hard masks
+    pub soft_masks_total_length: usize,
+}
 
-            sequences.insert(name, seq_offset);
+impl SequenceRecord {
+    pub fn info(&self) -> SequenceInfo {
+        SequenceInfo {
+            chr: self.chr.clone(),
+            length: self.length,
+            hard_masks_total_length: self.blocks_n.count(),
+            soft_masks_total_length: self.blocks_soft_mask.count(),
         }
-
-        let path = path.as_ref().to_path_buf();
-        Ok(TwoBitFile {
-            path,
-            sequences,
-            softmask_enabled,
-        })
     }
+}
 
-    /// Get the sizes of chromosomes in a 2bit file as a `HashMap`
-    pub fn chroms(&self) -> HashMap<String, Field> {
-        let mut result = HashMap::new();
-        for chr in self.sequences.keys() {
-            let seq = self.sequence_record(chr).expect("Chromosome must be valid");
-            result.insert((*chr).clone(), seq.dna_size);
+/// Summary information about a 2bit file.
+#[derive(Debug, Clone, Copy, PartialEq)]
+pub struct TwoBitFileInfo {
+    /// File size in bytes.
+    pub file_size: u64,
+    /// Number of chromosomes or sequences.
+    pub num_chromosomes: usize,
+    /// Total number of nucleotides over all sequences.
+    pub total_sequence_length: usize,
+    /// Total length of all hard masks in nucleotides.
+    pub hard_masks_total_length: usize,
+    /// Total length of all soft masks in nucleotides.
+    pub soft_masks_total_length: usize,
+}
+
+impl TwoBitFileInfo {
+    pub(crate) const fn new(file_size: u64) -> Self {
```

Comment:
Out of curiosity: why the `const`?

Reply:
Ah, that's to shut clippy up - it asks you to make things const if they can be const. Otherwise you'd have to either add a crate-wide lint ignore (which isn't nice because in some cases const methods may make sense), or add a method-specific lint ignore (which looks worse than just adding a const).